src/Type/Schema.php

```diff
@@ -341,6 +341,10 @@ private function loadType(string $typeName) : ?Type
 
         $type = $typeLoader($typeName);
 
+        if (! $type instanceof Type && is_callable($type)) {
+            $type = $type();
+        }
```

Comment:
```suggestion
        if (! $type instanceof Type) {
            if (is_callable($type) {
                $type = $type();
                if (! $type instanceof Type) {
                   $this->throwNotAType($typeName, $type);
                }
            } else {
                $this->throwNotAType($typeName, $type);
            }
        }
        
        // Remove the extra $type instanceof Type check below
```

This avoid one redundant call to `$type instanceof Type` in the happy path where the typeloader just returns a type.

Reply:
SGTM. 09109bc